packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -0,0 +1,726 @@
+  ///
+  /// When the callback is null, the entire segmented button is disabled,
+  /// and will not respond to input.
+  final void Function(Set<T>)? onSelectionChanged;
+
+  /// Determines if multiple segments can be selected at one time.
+  ///
+  /// If true, more than one segment can be selected. When selecting a
+  /// segment, the other selected segments will stay selected. Selecting an
+  /// already selected segment will unselect it.
+  ///
+  /// If false, only one segment will be selected at a time. When a segment
+  /// is selected, any previously selected segment will be unselected.
+  final bool multiSelectEnabled;
+
+  /// Determines if having no selected segments is allowed.
+  final bool emptySelectionAllowed;
+
+  /// Customizes this button's appearance.
+  ///
+  /// The following style properties apply to the entire segmented button:
+  ///
+  ///   * [ButtonStyle.shadowColor]
+  ///   * [ButtonStyle.elevation]
+  ///   * [ButtonStyle.padding]
+  ///   * [ButtonStyle.minimumSize]
+  ///   * [ButtonStyle.maximumSize]
+  ///   * [ButtonStyle.fixedSize]
+  ///   * [ButtonStyle.side] - which is used for both the shape and dividers
+  ///       between segments.
+  ///   * [ButtonStyle.shape]
+  ///
+  /// The following style properties are applied to each of the invidual
+  /// segment's state and will be resolved with a [MaterialState.selected]
+  /// if the segment is selected:
+  ///
+  ///   * [ButtonStyle.textStyle]
+  ///   * [ButtonStyle.backgroundColor]
+  ///   * [ButtonStyle.foregroundColor]
+  ///   * [ButtonStyle.overlayColor]
+  ///   * [ButtonStyle.surfaceTintColor]
+  ///   * [ButtonStyle.elevation]
+  ///   * [ButtonStyle.iconColor]
+  ///   * [ButtonStyle.iconSize]
+  ///   * [ButtonStyle.mouseCursor]
+  ///   * [ButtonStyle.visualDensity]
+  ///   * [ButtonStyle.tapTargetSize]
+  ///   * [ButtonStyle.animationDuration]
+  ///   * [ButtonStyle.enableFeedback]
+  ///   * [ButtonStyle.alignment]
+  ///   * [ButtonStyle.splashFactory]
+  final ButtonStyle? style;
+
+  /// Determines if the [selectedIcon] (usually a checkmark) is displayed on
+  /// the selected segments.
+  ///
+  /// If true the [selectedIcon] will be displayed at the start of the segment.
+  /// If both the [ButtonSegment.label] and [ButtonSegment.icon] is provided,
+  /// then the icon will be replaced with the [selectedIcon]. If only the icon
+  /// or the label is present then the [selectedIcon] will be shown at the start
+  /// of the segment.
+  ///
+  /// If false, then the [selectedIcon] is not used and will not be displayed
+  /// on selected segments.
+  final bool showSelectedIcon;
+
+  /// An icon that is used to indicate a segment is selected.
+  ///
+  /// If [showSelectedIcon] is true then for selected segments this icon
+  /// will be shown before the [ButtonSegment.label], replacing the
+  /// [ButtonSegment.icon] if it is specified.
+  ///
+  /// Defaults to [Icons.check].
+  final IconData? selectedIcon;
```

Comment:
Could this be a Widget instead?

Reply:
Possibly. This value can also be specified in the `SegmentedButtonThemeData` and I was a little concerned about putting a Widget into a theme data structure.

Reply:
@HansMuller do you have any issue with use a Widget in a theme data field?  I don't think we have done that before, is there a good reason not to?

Reply:
I don't think there's any reason not to. You'll have to configure the icon widget indirectly with IconTheme and you will not have control over the icon widget's size. Using a widget might be convenient for apps that just want to add a little padding. OTOH, using a widget enables a bunch of use cases, like icons that handle input, which we probably don't care about. I don't feel strongly about this one way or the other.

Reply:
Sounds good. I switched it to a `Widget`.